### PR TITLE
Community care

### PR DESCRIPTION
**Pull request by JosieSteele**
\\nssstats01\LIST_analytics\West Hub\02 - Scaled Up Work\RMarkdown\Locality Profiles\Unscheduled Care\testing output

File to check- Unscheduled_care_tocheck

### CODE REVIEW

File: Unscheduled Care/2. Unscheduled Care outputs.R
Comment (Moohan) [inline Unscheduled Care/2. Unscheduled Care outputs.R:13]: Leave this commented as it should only be needed when testing.
```suggestion
# lp_path <- "/conf/LIST_analytics/West Hub/02 - Scaled Up Work/RMarkdown/Locality Profiles/"
```

File: Unscheduled Care/2. Unscheduled Care outputs.R
Comment (Moohan) [inline Unscheduled Care/2. Unscheduled Care outputs.R:36]:
```suggestion
# LOCALITY <- "Inverness"
```

File: Unscheduled Care/2. Unscheduled Care outputs.R
Comment (Moohan) [inline Unscheduled Care/2. Unscheduled Care outputs.R:345]:
```suggestion
  emergency_adm_areas,
  financial_year == min(financial_year),
    location == LOCALITY,
    area_type == "Locality"
```

File: Unscheduled Care/2. Unscheduled Care outputs.R
Comment (Moohan) [inline Unscheduled Care/2. Unscheduled Care outputs.R:354]: Just delete code rather than commenting if it's no longer needed.
```suggestion
  mutate(data2 = format(data, big.mark = ","))
```

File: Unscheduled Care/2. Unscheduled Care outputs.R
Comment (Moohan) [inline Unscheduled Care/2. Unscheduled Care outputs.R:357]: Could you do some object renaming to make this clearer please? Particularly `data` and `data2` could be more specific / clear.

File: Unscheduled Care/2. Unscheduled Care outputs.R
Comment (Moohan) [inline Unscheduled Care/2. Unscheduled Care outputs.R:370]:
```suggestion
  mutate(data2 = format(data, big.mark = ","))
```

File: Unscheduled Care/2. Unscheduled Care outputs.R
Comment (Moohan) [inline Unscheduled Care/2. Unscheduled Care outputs.R:377]:
```suggestion
first_fy_hscp <- filter(
emergency_adm_areas,
  financial_year == min(financial_year),
    area_type == "HSCP"
    )$data
```

File: Unscheduled Care/2. Unscheduled Care outputs.R
Comment (Moohan) [inline Unscheduled Care/2. Unscheduled Care outputs.R:394]:
```suggestion
first_fy_scot <- filter(
emergency_adm_areas,
  financial_year == min(financial_year),
    location == "Scotland"
    )$data
```

File: Unscheduled Care/2. Unscheduled Care outputs.R
Comment (Moohan) [inline Unscheduled Care/2. Unscheduled Care outputs.R:412]:
```suggestion
first_fy_hb <- filter(
emergency_adm_areas,
                        financial_year == min(financial_year),
                          location == HB
                          )$data
```

File: Unscheduled Care/2. Unscheduled Care outputs.R
Comment (Moohan) [inline Unscheduled Care/2. Unscheduled Care outputs.R:439]:
```suggestion
  filter(
    year == max(year), 
    data == max(data)
  ) %>%
```

File: Unscheduled Care/2. Unscheduled Care outputs.R
Comment (Moohan) [inline Unscheduled Care/2. Unscheduled Care outputs.R:461]:
```suggestion
    year == max(year),
    data == min(data)
```

File: Unscheduled Care/2. Unscheduled Care outputs.R
Comment (Moohan) [inline Unscheduled Care/2. Unscheduled Care outputs.R:532]:
```suggestion
  financial_year == min(financial_year),
    location == LOCALITY,
     area_type == "Locality"
     )$data
```

File: Unscheduled Care/2. Unscheduled Care outputs.R
Comment (Moohan) [inline Unscheduled Care/2. Unscheduled Care outputs.R:537]:
```suggestion
  mutate(data2 = format(data, big.mark = ",")) 
```

File: Unscheduled Care/2. Unscheduled Care outputs.R
Comment (Moohan) [inline Unscheduled Care/2. Unscheduled Care outputs.R:312]: `drop_na()` is from tidyr and is a slightly nicer way to achieve the same result. 
```suggestion
  drop_na(year)
```

File: Unscheduled Care/2. Unscheduled Care outputs.R
Comment (Moohan) [inline Unscheduled Care/2. Unscheduled Care outputs.R:359]: This crops up a lot and might benefit from being a function.

```r
good_function_name <- function(numerator, denominator) {
    round(abs(numerator - denominator) / denominator* 100, digits = 1)
}
```

```suggestion
percent_rate_change <- good_function_name(latest_emergency_adm_loc2, first_fy_rate)
```

File: Unscheduled Care/2. Unscheduled Care outputs.R
Comment (Moohan) [inline Unscheduled Care/2. Unscheduled Care outputs.R:361]:
```r
function_name <- function(latest, first) {
    case_match(
        near(latest, first) ~ "same",
        latest > first ~ "increase",
        latest < first ~ "decrease"
    )
}
```

File: Unscheduled Care/2. Unscheduled Care outputs.R
Comment (Moohan) [inline Unscheduled Care/2. Unscheduled Care outputs.R:1]: I started looking through this but was making very similar comments repeatedly, hopefully, my suggestions make sense, and if you agree can you replicate them throughout as appropriate? I've also made some suggestions around creating new helper functions, I'm not sure if these make sense as part of the USC script or whether they're general enough to be declared in the 'general script'

File: Master RMarkdown Document & Render Code/Locality_Profiles_Master_Markdown.Rmd
Comment (Moohan) [inline Master RMarkdown Document & Render Code/Locality_Profiles_Master_Markdown.Rmd:682]: Might be nicer to use `scales::percent` i.e. `r scales::percent(rate_chane)` instead of `r percent_rate_change`% This will handle the rounding and everything, so we then wouldn't need to do this in the R script and just let the formatting be done in the RMarkdown file.

File: Master RMarkdown Document & Render Code/Locality_Profiles_Master_Markdown.Rmd
Comment (Moohan) [inline Master RMarkdown Document & Render Code/Locality_Profiles_Master_Markdown.Rmd:681]:
```suggestion
As presented in Figure `r x`, the emergency admissions rate in the `r LOCALITY` locality for `r max_year_ea` is `r latest_emergency_adm_loc1`, a `r percent_rate_change`% `r word_change_rate` since `r min_year_ea`. The `r HSCP` HSCP rate is `r hscp_emergency_adm1`, a `r hscp_rate_change`% `r word_change_hscp` since `r min_year_ea`. The `r HB` health board rate is `r hscp_emergency_adm1` in `r max_year_ea`, a `r hb_rate_change`% `r word_change_hb` since `r min_year_ea` and the Scotland rate is `r scot_emergency_adm1`, a `r scot_rate_change`% `r word_change_scot` since `r min_year_ea`.
```

File: Master RMarkdown Document & Render Code/Locality_Profiles_Master_Markdown.Rmd
Comment (Moohan) [inline Master RMarkdown Document & Render Code/Locality_Profiles_Master_Markdown.Rmd:817]:
```suggestion
Figure `r x` presents the emergency admissions from falls rate per 100,000 population aged 65 and over in the `r LOCALITY` locality from `r min_year_falls` to `r max_year_falls`. 
```

File: Master RMarkdown Document & Render Code/Locality_Profiles_Master_Markdown.Rmd
Comment (Moohan) [inline Master RMarkdown Document & Render Code/Locality_Profiles_Master_Markdown.Rmd:838]:
```suggestion
Information on the conditions included in Potentially Preventable Admissions (PPA)s is available in Appendix 3. In `r max_fy`, **`r latest_ppa_65plus`%** of PPAs in `r LOCALITY` were amongst those aged 65 and over, and **`r latest_ppa_under65`%** were amongst those aged under 65. 

Figure `r x` presents the PPA rate per 100,000 population in the `r LOCALITY` locality from `r min_year_ppa_areas` to `r max_year_ppa_areas`. The rate per 100,000 population for PPAs in the `r LOCALITY` locality for `r max_year_ppa_areas ` is `r latest_ppa_loc$data2[2]`, a `r ppa_diff`% `r ppa_word_change` since `r latest_ppa_loc$financial_year[1]` and the `r HSCP` HSCP rate is `r hscp_ppa$data[2]`, a `r ppa_diff_hscp`% `r ppa_word_change_hscp` since `r latest_ppa_loc$financial_year[1]`.
The `r HB` health board rate for `r hb_ppa$financial_year[2]` is `r hb_ppa$data2[2]`, a `r diff_hb_ppa`% `r word_change_hb_ppa` since `r hb_ppa$financial_year[]` and the Scotland rate is `r scot_ppa$data2[2]`, a `r diff_scot_ppa`% `r  word_change_scot_ppa` since `r hb_ppa$financial_year[1]`.
```

File: Master RMarkdown Document & Render Code/Locality_Profiles_Master_Markdown.Rmd
Comment (Moohan) [inline Master RMarkdown Document & Render Code/Locality_Profiles_Master_Markdown.Rmd:918]:
```suggestion
1. Population projections are not currently provided by the National Records Scotland (NRS) at the locality level. To explore how the population in `r LOCALITY` is expected to change in the future, the percent changes in population projection to 2025 for `r HSCP` by age group and gender were calculated using the NRS Local Authority Population Projections. These percent changes were then applied to the `r LOCALITY` 2018 mid-year population estimates (also split by age group and gender) to obtain population projection estimates for `r LOCALITY`, based on the projections for the HSCP and the current population structure of the locality.
```

File: Master RMarkdown Document & Render Code/Locality_Profiles_Master_Markdown.Rmd
Comment (Moohan) [inline Master RMarkdown Document & Render Code/Locality_Profiles_Master_Markdown.Rmd:924]: Pretty sure this link is no longer valid and there is a new PHS website one.

File: Master RMarkdown Document & Render Code/Locality_Profiles_Master_Markdown.Rmd
Comment (Moohan) [inline Master RMarkdown Document & Render Code/Locality_Profiles_Master_Markdown.Rmd:920]:
```suggestion
2. Care Home data included in the Services Map and Table was sourced from the [Care Inspectorate](https://www.careinspectorate.com/images/documents/5713/MDSF_data_31%20May%202020.csv). [GP Practice](https://www.opendata.nhs.scot/dataset/gp-practice-contact-details-and-list-sizes) data from October 2021, and [Hospital](https://www.opendata.nhs.scot/dataset/hospital-codes) and [A&E](https://www.opendata.nhs.scot/dataset/nhs-scotland-accident-emergency-sites) data was sourced from Public Health Scotland Open Data platform. Only services that are within the physical boundary of the HSCP or Locality are included in the map and table, so there may be services outside `r HSCP` that people may use but are not shown. Information on access deprivation was taken from [ScotPHO](https://scotland.shinyapps.io/ScotPHO_profiles_tool/).
```

**Review by Moohan — CHANGES_REQUESTED** [review]

File: Unscheduled Care/2. Unscheduled Care outputs.R
Comment (JosieSteele) [reply Unscheduled Care/2. Unscheduled Care outputs.R:361]: Using case_when instead

word_change_calc <- function(latest, first) {
  case_when(
    near(latest, first) ~ "same",
    latest > first ~ "increase",
    latest < first ~ "decrease"
  )
}

File: Unscheduled Care/2. Unscheduled Care outputs.R
Comment (JosieSteele) [reply Unscheduled Care/2. Unscheduled Care outputs.R:452]: Couldn't get this to work as I need it to filter year and then data. Think this tries both at same time?

File: Unscheduled Care/2. Unscheduled Care outputs.R
Comment (CliveWG) [reply Unscheduled Care/2. Unscheduled Care outputs.R]: I just wondered how this fits in with the actual markdown text, for example, when I changed it for Inverclyde, I had to change the values to "same as", "increase of", "decrease of" and adjust the markdown text accordingly?

File: Unscheduled Care/2. Unscheduled Care outputs.R
Comment (CliveWG) [reply Unscheduled Care/2. Unscheduled Care outputs.R:452]: Yes, you're spot on, this is the equivalent of max(year) AND max(data). Unless those more technically astute than I know some tricks, the command might have to stay as original (rows 448 to 452).

**Review by Moohan — CHANGES_REQUESTED** [review]

File: Unscheduled Care/2. Unscheduled Care outputs.R
Comment (Moohan) [inline Unscheduled Care/2. Unscheduled Care outputs.R:1580]: @JosieSteele - Some conflict markers were introduced (it's probably because I'd made some commits which you hadn't pulled... sorry!)

File: Master RMarkdown Document & Render Code/Locality_Profiles_Master_Markdown.Rmd
Comment (Moohan) [inline Master RMarkdown Document & Render Code/Locality_Profiles_Master_Markdown.Rmd:683]: This is quite a petty suggestion 😆 but if the rate change value starts with 8 or 18 the article should be 'an' not 'a'.

For example "a 17.1% increase ..." is fine but "a 18.1% ..." or "a 82.5% ..." should be "an 18.1% ..." or "an 82.5% ...".

This is probably an issue throughout the profiles, but we can tackle it here and worry about the rest 
later!

I suggest adding a new function (probably in General, for ease later).

```r
get_article <- function(number) {
  if (substr(number, 1, 1) == "8" || substr(number, 1, 2) == "18") {
    return("an")
  } else {
    return("a")
  }
}
```

Then update the text in the Markdown to:
```suggestion
As presented in Figure `r x`, the emergency admissions rate in the `r LOCALITY` locality for `r max_year_ea` is `r latest_emergency_adm_loc1`, `r get_article(percent_rate_change)` `r percent_rate_change`% `r word_change_rate` since `r min_year_ea`. The `r HSCP` HSCP rate is `r hscp_emergency_adm1`, `r get_article(hscp_rate_change)` `r hscp_rate_change`% `r word_change_hscp` since `r min_year_ea`. The `r HB` health board rate is `r hscp_emergency_adm1` in `r max_year_ea`, `r get_article(hb_rate_change)` `r hb_rate_change`% `r word_change_hb` since `r min_year_ea` and the Scotland rate is `r scot_emergency_adm1`, `r get_article(scot_rate_change)` `r scot_rate_change`% `r word_change_scot` since `r min_year_ea`.
```

Even writing this comment was a chunk of work, so possibly we just move this to an issue to be tackled as a separate task, so that it doesn't hold up the wider Community Care chapter improvements.

Comment (Moohan) [reply Master RMarkdown Document & Render Code/Locality_Profiles_Master_Markdown.Rmd:683]: This now a PR to this branch (#111) as well as a PR to apply it to the wider project (#110)

**Comment by JosieSteele** [comment]
Full profile available in: \\nssstats01\LIST_analytics\West Hub\02 - Scaled Up Work\RMarkdown\Locality Profiles\Unscheduled Care\testing output
Angus North East

**Review by Moohan — DISMISSED** [review]

**Comment by Moohan** [comment]
I just checked through the output and all looks good to me. Happy for you to merge @JosieSteele, but ideally someone else could have a look and leave a review/approval!

**Comment by JosieSteele** [comment]
Temp fix to figure 33. Other issues will be added to issues page

**Review by Moohan — APPROVED** [review]

**Pull request merged**
